packages/pharos/src/components/tabs/pharos-tabs.ts

```diff
@@ -24,40 +27,35 @@ export class PharosTabs extends PharosElement {
   @property({ type: Boolean, reflect: true, attribute: 'panel-separator' })
   public panelSeparator = false;
 
+  @queryAssignedElements({ selector: _allTabsSelector })
+  private _tabs!: NodeListOf<PharosTab>;
+
   public static override get styles(): CSSResultArray {
     return [tabsStyles];
   }
 
-  protected override firstUpdated(): void {
-    this.addEventListener('pharos-tab-selected', this._handleTabSelected);
+  protected override async firstUpdated(): Promise<void> {
+    this.addEventListener('pharos-tab-selected', (e) =>
+      this._handleTabSelected(e.target as PharosTab)
+    );
     this.addEventListener('keydown', this._handleKeydown);
     this.addEventListener('focusout', this._handleFocusout);
 
-    const tabs: NodeListOf<PharosTab> = this.querySelectorAll(
-      `[data-pharos-component="PharosTab"]`
-    );
-
-    tabs.forEach((tab) => {
+    for (const tab of this._tabs) {
       const panel = this._queryPanelByTab(tab);
-
       tab.setAttribute('aria-controls', panel?.id || '');
       panel?.setAttribute('aria-labelledby', tab.id);
-    });
+      await tab.updateComplete;
+      await panel?.updateComplete;
+    }
 
-    this._selectInitialTab(tabs);
+    this._selectInitialTab();
   }
 
-  private _selectInitialTab(tabs: NodeListOf<PharosTab>): void {
-    const selected: PharosTab | null = this.querySelector(
-        `[data-pharos-component="PharosTab"][selected]`
-      ),
-      selectedTab: PharosTab = selected ? selected : tabs[0],
-      selectedPanel: PharosTabPanel | null = this._queryPanelByTab(selectedTab);
-
-    selectedTab.selected = true;
-    if (selectedPanel) {
-      selectedPanel.selected = true;
-    }
+  private _selectInitialTab(): void {
+    const selected: PharosTab | null = this.querySelector(`${_allTabsSelector}[selected]`);
+    const selectedTab: PharosTab = selected ? selected : this._tabs[0];
+    this._handleTabSelected(selectedTab);
   }
```

Comment:
Simplified this function by adjusting `_handleTabSelected` to work agnostic of how/why the tab got selected, instead of managing different cases of selection different ways.